classes/network/Connection.py

```diff
@@ -0,0 +1,310 @@
+import multiprocessing
+
+from multiprocessing.queues import Empty
+
+from classes.utils.Utils import Version
+from quarry.net import server
+from threading import Thread
+from twisted.internet import reactor
+
+from .PacketType import BasicNetwork, PacketType
+from .versions.v578 import v1_15_2
+
+
+# TODO Change that
+entity_id = 1
+QUEUE_SIZE = 100000
+
+
+class PlayerNetwork(server.ServerProtocol):
+
+    def handle_loop(self):
+        """
+        This method will check if there is a packet to send to the player and if there is, it sends the packet\n
+        This method is called 20 times per seconds
+        """
+        while True:
+            try:
+                item = self.TASK_QUEUE.get_nowait()
+            except (KeyboardInterrupt, Empty):
+                return
+            if item:
+                # Handle it
+                packet_type = item['packet_type']
+                data = item['data']
+                self.make_packet_and_send(packet_type=packet_type, data=data)
+            else:
+                break
+
+    def player_joined(self):
+        super(PlayerNetwork, self).player_joined()
+
+        if str(self.protocol_version) not in self.factory._protocol:
+            # Version not supported
+            self.close('Invalid protocol or not supported, please use a supported version of Minecraft !')
+            return
+        self._protocol = self.factory._protocol[str(self.protocol_version)]
+        self.version = Version.get_version(self.protocol_version)
+
+        self.TASK_QUEUE = multiprocessing.Queue(QUEUE_SIZE)
+        self._handle_loop = self.ticker.add_loop(1, self.handle_loop)
+
+        # TODO Generate it
+        global entity_id
+        self.entity_id = entity_id
+        entity_id = entity_id + 1
```

Comment:
I don't quite like how that is executed, particularly the fact that we are declaring a global here, although I might be too stupid to realize that that's how you use globals

Reply:
That'll be also moved in another place later, generaly I never use global variables